From: Lorenzo Bianconi <lorenzo@kernel.org>
Date: Mon, 8 Mar 2021 23:01:49 +0100
Subject: [PATCH] mac80211: set sk_pacing_shift for 802.3 txpath

Similar to 802.11 txpath, set socket sk_pacing_shift for 802.3 tx path.

Signed-off-by: Lorenzo Bianconi <lorenzo@kernel.org>
---

--- a/net/mac80211/tx.c
+++ b/net/mac80211/tx.c
@@ -4173,6 +4173,9 @@ static bool ieee80211_tx_8023(struct iee
 	unsigned long flags;
 	int q = info->hw_queue;
 
+	if (sta)
+		sk_pacing_shift_update(skb->sk, local->hw.tx_sk_pacing_shift);
+
 	if (ieee80211_queue_skb(local, sdata, sta, skb))
 		return true;
 
